### ISSUE TITLE
Kubemark-500 is broken

### ISSUE THREAD

**Issue by wojtek-t** [kubernetes/kubernetes#46196]
It is failing to create all the nodes since:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-kubemark-500-gce/5030

@kubernetes/sig-scalability-bugs @gmarek @shyamjvs

**Comment by wojtek-t** [comment]
It seems that this PR was the only one that was merged when it started failing:
https://github.com/kubernetes/kubernetes/pull/45933
@smarterclayton @deads2k - not sure what the reason actually is though

What is suspicious to me is that is seems that in **all** runs exactly 490 hollow nodes out of 500 started.


It seems that something has changed around scheduling of nodes, because there are always 10 pending hollow nodes in this test. And looking into scheduler logs, those are not scheduling due to unsufficient CPU:
```
I0522 06:58:24.373109       7 event.go:218] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kubemark", Name:"hollow-node-63p8r", UID:"96c0cb1a-3eba-11e7-8e51-42010a800002", APIVersion:"v1", ResourceVersion:"2548", FieldPath:""}): type: 'Warning' reason: 'FailedScheduling' No nodes are available that match all of the following predicates:: Insufficient cpu (6).
```

**Comment by wojtek-t** [comment]
OK - now I'm pretty sure that this PR breaks this, because it also cause schedulerpredicates tests to fail:
https://k8s-testgrid.appspot.com/google-gce#gce-serial

Though, I'm looking into this PR and to be honest I don't know what is wrong with it yet.

**Comment by wojtek-t** [comment]
the revert seems to solve the problem. So I'm closing this one.

**Comment by smarterclayton** [comment]
It's probably someone mutating the shared cache.

**Comment by ncdc** [comment]
The per-PR tests would fail if that's the case (the mutation detector should be enabled for them).

**Comment by wojtek-t** [comment]
> It's probably someone mutating the shared cache.

It doesn't seem like that - in that case it would be more like a race.
The behavior in kubemark-500 was super stable - we were always starting exactly 490 out of 500 hollow nodes.

**Comment by shashidharatd** [comment]
https://k8s-testgrid.appspot.com/cluster-federation#gce
Above is another instance of failing test case fixed by this reverting PR #46199
The test case started failing when the commit https://github.com/kubernetes/kubernetes/commit/479f01d340b917ac3544f094f38edaed0ab44128 was merged.

**Comment by smarterclayton** [comment]
Heh - turns out the watch cache behaves differently than the underlying storage watch for delete events.  My filtering function is written for the underlying storage behavior.  The behavior in cache looks like a bug we never caught.
